src/mapping/impl/BasisFunctions.hpp

```diff
@@ -1,11 +1,43 @@
 #pragma once
 
+#if defined(__NVCC__)
+
+#include <cuda_runtime.h>
+#define BOOST_PP_VARIADICS 1
+#define PRECICE_HOST_DEVICE __host__ __device__
+#define PRECICE_MEMORY_SPACE __device__
+#define FMA fma
+
+#elif defined(__HIPCC__)
+
+// #define __HIP_PLATFORM_AMD__
+#include <hip/hip_runtime.h>
+#define PRECICE_HOST_DEVICE __host__ __device__
+#define PRECICE_MEMORY_SPACE __device__
+#define FMA fma
+
+#else
+
+#define PRECICE_HOST_DEVICE
+#define PRECICE_MEMORY_SPACE
+#define FMA std::fma
+
+#endif
+
+PRECICE_MEMORY_SPACE const double NUMERICAL_ZERO_DIFFERENCE = 1.0e-14;
+
 #include "logging/Logger.hpp"
 #include "math/math.hpp"
 
 namespace precice {
 namespace mapping {
 
+struct RadialBasisParameters {
+  double parameter1;
+  double parameter2;
+  double parameter3;
+};
```

Comment:
We probably need some documentation here. Unclear to me what these parameters mean exactly.

Reply:
In this form, they don't have an exact meaning apart from being RBF parameters (shape, radius, etc.). Initally, I wanted to move RBF parameters to the GPU using `std::array` that is part of each RBF kernel class. However, AMD does not support this container on its GPUs because of `constexpr` of the `.at()` method. Hence, each RBF kernel fills 0-3 parameters in this struct, which is then given to the RBF kernel lambda that is executed on the GPU. Having it this way, we can have one single evaluation device kernel on both Nvidia and AMD by always passing this parameter struct to the GPU kernel. Does this clarify it?

Reply:
yes, thanks :pray: 
but let's at this documentation also in the source code

Reply:
Hi, sorry for the delay, but I added a docstring here: https://github.com/Timo-Schrader/precice/tree/patch-1

Can open a PR if you are fine with that version. Let me know if you need anything else!

Reply:
As far as I can say from now, I will wrap up the whole PR next week and then cherry-pick your changes. Thanks!